### PR TITLE
Code Quality: Remove `shoppingIntegration` feature flag code 

### LINKED ISSUE #11793: Code Quality: Remove `shoppingIntegration` feature flag code
## Task Description

Now that this has been enabled by default in the last release we can remove the relevant code bits.

### PR DESCRIPTION
**Pull request by timarney**
## Context

> Now that this has been enabled by default in the last release we can remove the relevant code bits.

## Summary

Removes shoppingIntegration feature flag.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11793

### CODE REVIEW

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js [hunk 1]
```diff
@@ -162,8 +162,7 @@ function ShoppingPane(props) {
     [debouncedProductsQuery, order, orderby, searchTerm]
   );
 
-  const isShoppingEnabled =
-    'none' !== shoppingProvider && isShoppingIntegrationEnabled;
+  const isShoppingEnabled = 'none' !== shoppingProvider;
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js:163]: The lint error on this file is correct.

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js [hunk 2]
```diff
@@ -173,12 +170,12 @@ function ShoppingPane(props) {
   }, [isShoppingEnabled]);
 
   useEffect(() => {
-    if (!isShoppingEnabled) {
+    if (!isShoppingEnabled && products.length === 0) {
```
Comment (timarney) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js]: Added check for `products.length` here

When I was testing this was seeing issues with state updates in the useEffect

Screen Shot 2022-06-27 at 6 44 33 AM

File: packages/story-editor/src/components/library/panes/shopping/shoppingPane.js [hunk 3]
```diff
@@ -173,12 +170,12 @@ function ShoppingPane(props) {
...
       setProducts(currentPageProducts?.map(({ product }) => product));
       setIsLoading(false);
       setLoaded(true);
     }
-  }, [currentPageProducts, isShoppingEnabled]);
+  }, [currentPageProducts, isShoppingEnabled, products.length]);
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/library/panes/shopping/shoppingPane.js]:
```suggestion
  }, [currentPageProducts, isShoppingEnabled, products]);
```

Comment (swissspidy) [reply packages/story-editor/src/components/library/panes/shopping/shoppingPane.js]: If we're only interested in the length of the array it's better to just pass that and not the whole array.

```js
const numProducts = products.length;
useEffect(() => {
  // ... Do something with numProducts
}, [numProducts]);
```

**Comment by swissspidy** [comment]
Since there are some more code changes, would be good to get Felipe's eyes on it through QA

**Review by swissspidy — APPROVED** [review]

**Pull request merged**